CHANGELOG.md

```diff
@@ -21,6 +21,7 @@ If you do have such headers configured, then you must update the header configur
 - server: add logs for action handlers
 - server: add request/response sizes in event triggers (and scheduled trigger) logs (#5463)
 - server: change startup log kind `db_migrate` to `catalog_migrate` (#5531)
+- server: don't throw validation error when creating remote relationships joining singleton type with array type with same base type (fixes #5133)
```

Comment:
Does this really close 5133? I don't see any mention of list types there..

Reply:
Yes, it does. The issue doesn't mention anything about the list type, because the error message thrown is incorrect. Ideally, the error message should have been 

```Error remote relationship: cannot validate remote relationship because expected type \"[Int]\" but got \"Int\" ```

instead of 

```Error remote relationship: cannot validate remote relationship because expected type \"Int\" but got \"Int\"```

The reason the list notation is omitted from the error message because of this [line](https://github.com/hasura/graphql-engine/blob/ea76e7840eaf2da68c7170a3d55cccf60083954b/server/src-lib/Hasura/RQL/DDL/RemoteRelationship/Validate.hs#L68)

This PR solves that issue.

Reply:
This error message can occur in another case too, while joining an `[Int]` type to a `Int` type, where the validation is justified but the error message will be the same 

```Error remote relationship: cannot validate remote relationship because expected type \"Int\" but got \"Int\"``` 

which is not very helpful. 

I'll fix the error message.

Reply:
> Also, is this deficient in other ways according to the spec?

Yes, it was. My patch would have allowed to make a remote relationship joining types `[[Int]]` and `[[[Int]]]`, which is illegal, as specified in the GraphQL spec.

I have fixed this in the next commit